packages/base/src/Boot.ts

```diff
@@ -7,56 +8,55 @@ import { registerCurrentRuntime } from "./Runtimes.js";
 import { getFeature } from "./FeaturesRegistry.js";
 import type OpenUI5Support from "./features/OpenUI5Support.js";
 import type F6Navigation from "./features/F6Navigation.js";
-import { PromiseResolve } from "./types.js";
 
-let bootPromise: Promise<void>;
+const eventProvider = new EventProvider<void, void>();
+let booted = false;
 
 /**
- * Attach  a callback that will be executed on boot
+ * Attaches a callback that will be executed after boot finishes.
  * @public
- * @param listener
+ * @param { Function } listener
  */
-const attachBoot = async (listener: () => void) => {
-	await boot();
+const attachBoot = (listener: () => void) => {
+	if (!booted) {
+		eventProvider.attachEvent("boot", listener);
+		return;
+	}
+
 	listener();
 };
 
 const boot = async (): Promise<void> => {
-	if (bootPromise !== undefined) {
-		return bootPromise;
+	if (booted) {
+		return;
 	}
 
-	const bootExecutor = async (resolve: PromiseResolve) => {
-		if (typeof document === "undefined") {
-			resolve();
-			return;
-		}
-		registerCurrentRuntime();
-
-		const openUI5Support = getFeature<typeof OpenUI5Support>("OpenUI5Support");
-		const isOpenUI5Loaded = openUI5Support ? openUI5Support.isLoaded() : false;
-		const f6Navigation = getFeature<typeof F6Navigation>("F6Navigation");
+	if (typeof document === "undefined") {
+		return;
+	}
 
-		if (openUI5Support) {
-			await openUI5Support.init();
-		}
+	registerCurrentRuntime();
 
-		if (f6Navigation && !isOpenUI5Loaded) {
-			f6Navigation.init();
-		}
+	const openUI5Support = getFeature<typeof OpenUI5Support>("OpenUI5Support");
+	const isOpenUI5Loaded = openUI5Support ? openUI5Support.isLoaded() : false;
+	const f6Navigation = getFeature<typeof F6Navigation>("F6Navigation");
 
-		await whenDOMReady();
-		await applyTheme(getTheme());
-		openUI5Support && openUI5Support.attachListeners();
-		insertFontFace();
-		insertSystemCSSVars();
+	if (openUI5Support) {
+		await openUI5Support.init();
+	}
 
-		resolve();
-	};
+	if (f6Navigation && !isOpenUI5Loaded) {
+		f6Navigation.init();
+	}
 
-	bootPromise = new Promise(bootExecutor as (resolve: PromiseResolve) => void);
+	await whenDOMReady();
+	await applyTheme(getTheme());
+	openUI5Support && openUI5Support.attachListeners();
+	insertFontFace();
+	insertSystemCSSVars();
 
-	return bootPromise;
+	await eventProvider.fireEventAsync("boot");
+	booted = true;
```

Comment:
I suggest to set the state first and then fire the event. Otherwise someone could execute some logic in a `boot` handler that could come back to this module and incorrectly assume `booted` is false.

Reply:
Agree